coldfront/core/user/utils.py

```diff
@@ -400,3 +402,22 @@ def needs_host(user):
     else:
         # Check if the user has a host user already.
         return user.userprofile.host_user is None
+
+
+def get_compute_resources_for_user(user):
+    """Returns a set of the compute resources for all projects a user
+    belongs to."""
+    proj_users = ProjectUser.objects.filter(user=user, status__name='Active')
+
+    compute_resources = set()
+    for proj_user in proj_users:
+        allocations = Allocation.objects.filter(
+            project=proj_user.project,
+            status__name='Active',
+            resources__name__icontains='Compute')
+
+        for allocation in allocations:
+            for resource in allocation.resources.all():
```

Comment:
Just loop over the compute ones. (Can't think of a case now where a compute `Allocation` would have other resources, but it could occur in the future.)